docs/topics/multiplatform/multiplatform-run-tests.md

```diff
@@ -1,67 +1,375 @@
+
+Here is an example of the output generated by a successful test run:
+
+![Test output](run-test-results.png){width=700}
+
+## Work with more complex projects
+
+### Write tests for common code
+
+You've already created a test for common code with the `grep()` function. Now let's consider a more advanced common code
+test with the `CurrentRuntime` class. This class contains details of the platform on which the code is executed.
+For example, it might have the values "OpenJDK" and "17.0" for Android unit tests that run on a local JVM.
+
+An instance of `CurrentRuntime` should be created with the name and version of the platform as strings, where the
+version is optional. When the version is present, you only need the number at the start of the string, if available.
+
+1. In the `commonMain/kotlin` folder, create a new `org.kmp.testing` directory.
+2. In this directory, create the `CurrentRuntime.kt` file and update it with the following implementation:
+
+    ```kotlin
+    class CurrentRuntime(val name: String, rawVersion: String?) {
+        companion object {
+            val versionRegex = Regex("^[0-9]+(\\.[0-9]+)?")
+        }
+    
+        val version = parseVersion(rawVersion)
+    
+        override fun toString() = "$name version $version"
+    
+        private fun parseVersion(rawVersion: String?): String {
+            val result = rawVersion?.let { versionRegex.find(it) }
+            return result?.value ?: "unknown"
+        }
     }
-}
+    ```
+
+3. In the `commonTest` folder, create a new `org.kmp.testing` directory.
+4. In this directory, create the `CurrentRuntimeTest.kt` and update it with following platform and framework-agnostic test:
+
+    ```kotlin
+    import kotlin.test.Test
+    import kotlin.test.assertEquals
+
+    class CurrentRuntimeTest {
+        @Test
+        fun shouldDisplayDetails() {
+            val runtime = CurrentRuntime("MyRuntime", "1.1")
+            assertEquals("MyRuntime version 1.1", runtime.toString())
+        }
+    
+        @Test
+        fun shouldHandleNullVersion() {
+            val runtime = CurrentRuntime("MyRuntime", null)
+            assertEquals("MyRuntime version unknown", runtime.toString())
+        }
+    
+        @Test
+        fun shouldParseNumberFromVersionString() {
+            val runtime = CurrentRuntime("MyRuntime", "1.2 Alpha Experimental")
+            assertEquals("MyRuntime version 1.2", runtime.toString())
+        }
+    
+        @Test
+        fun shouldHandleMissingVersion() {
+            val runtime = CurrentRuntime("MyRuntime", "Alpha Experimental")
+            assertEquals("MyRuntime version unknown", runtime.toString())
+        }
+    }
+    ```
+
+You can run this test using any of the ways [available in the IDE](#run-tests).
+
+### Add platform-specific tests
+
+> Here, the mechanism of expected and actual declarations is used for brevity and simplicity. In more complex code, a
```

Comment:
Do you think it would  be helpful to link to: https://kotlinlang.org/docs/multiplatform-connect-to-apis.html here?